open_catalyst_dataset/adsorptions.py

```diff
@@ -70,20 +98,22 @@ def sample_structures(bulk_database='bulks.db', n_species_weights=None):
                             we sampled.
     '''
     # Choose which surface we want
+    print('choosing a slab...')
     n_species = choose_n_species(n_species_weights)
     elements = choose_elements(bulk_database, n_species)
     bulk = choose_bulk(bulk_database, elements)
-    surface = choose_surface(bulk)
+    unit_surface_slab = choose_surface(bulk)
+    slab = tile_atoms(unit_surface_slab)
+    surface_atoms = find_surface_atoms_indices(bulk, slab)
```

Comment:
This much detail should not be at the top function. The tiling and finding of surface atoms should be abstracted to the underlying functions.